adapter/pkg/synchronizer/apis_fetcher.go

```diff
@@ -223,20 +227,76 @@ func ReadDeployments(reader *zip.Reader) (*DeploymentDescriptor, error) {
 			f, err := file.Open()
 			if err != nil {
 				logger.LoggerSync.Errorf("Error reading deployment descriptor: %v", err)
-				return deploymentDescriptor, err
+				return deploymentDescriptor, nil, err
 			}
 			data, err := ioutil.ReadAll(f)
 			_ = f.Close() // Close the file here (without defer)
 			if err != nil {
 				logger.LoggerSync.Errorf("Error reading deployment descriptor: %v", err)
-				return deploymentDescriptor, err
+				return deploymentDescriptor, nil, err
 			}
 			logger.LoggerSync.Debugf("Parsing content of deployment descriptor, content: %s", string(data))
 			if err = json.Unmarshal(data, deploymentDescriptor); err != nil {
 				logger.LoggerSync.Errorf("Error parsing JSON content of deployment descriptor: %v", err)
-				return deploymentDescriptor, err
+				return deploymentDescriptor, nil, err
 			}
+		} else if len(environments) > 0 && strings.Contains(file.Name, envPropsFile) {
+			logger.LoggerSync.Debug("Reading environment specific properties of Environments")
+
+			f, err := file.Open()
+			if err != nil {
+				logger.LoggerSync.Errorf("Error reading environment specific properties: %v", err)
+				return deploymentDescriptor, nil, err
+			}
+			fileContent, err := ioutil.ReadAll(f)
+			_ = f.Close() // Close the file here (without defer)
+			if err != nil {
+				logger.LoggerSync.Errorf("Error reading environment specific properties: %v", err)
+				return deploymentDescriptor, nil, err
+			}
+			logger.LoggerSync.Debugf("Parsing content of environment specific properties, content: %s", string(fileContent))
+			apiEnvProps, err := parseEnvProps(fileContent)
+			if err != nil {
+				logger.LoggerSync.Errorf("Error occurred while parsing environment specific properties : %v : %v",
+					file.Name, err.Error())
+				return deploymentDescriptor, nil, err
+			} else if apiEnvProps != nil {
+				return deploymentDescriptor, apiEnvProps, nil
+			}
+		}
+
+	}
+	return deploymentDescriptor, nil, nil
+}
+
+func parseEnvProps(data []byte) (map[string]map[string]APIEnvProps, error) {
+	var envPropsFile map[string]interface{}
+	envProps := make(map[string]map[string]APIEnvProps)
+	if err := json.Unmarshal(data, &envPropsFile); err != nil {
+		logger.LoggerSync.Errorf("Error parsing Environment specific values: %v", err.Error())
+		return nil, err
+	}
+
+	if apisData, found := envPropsFile["apis"]; found {
+		if apis, ok := apisData.(map[string]interface{}); ok {
+			for apiUUID, apiData := range apis {
+				apiProps := make(map[string]APIEnvProps)
+				if api, ok := apiData.(map[string]interface{}); ok {
+					var envProps APIEnvProps
+					for envLabel, envData := range api {
+						if err := parser.Decode(envData, &envProps); err != nil {
+							logger.LoggerSync.Debugf("Error parsing environment specific values: %v", err.Error())
+							return nil, err
+						}
+						apiProps[envLabel] = envProps
+					}
+				}
+				envProps[apiUUID] = apiProps
```

Comment:
sorry for the delay after the first comment. Just thought of this. If this file sends data about many APIs, will there be a scenario we have to maintain them? As in, will there be an even where only the envs are sent without any APIs. Can that be the reason to send envProps that belongs to multiple APIs?

Reply:
This file is only coming from APIM internal rest API. This file contains values for all the APIs if they have some env values. That is the only requirement we have at the moment.

Reply:
This would also mean that every time an API event occurs, we will be looping through all the APIs, for all its envs, and will be creating maps, unmarshelling objects, and decoding.

Reply:
@AmaliMatharaarachchi how about using the method parseEnvProps after we get to know about the api uuid. then we can directly access the info we want and stop. the larger map can be discarded and only a smaller map will stay.

Reply:
@suksw The env_properties.json contains the environment label specific data only for the APIs corresponding to the API event. If I understand your concern correctly, so that won't be an issue since we only marshell env props of apis that we wants to update, am I right?

Reply:
Got the idea after the discussion. So the env_properties.json is located outside the larger zip (which contains all the API zips). Therefore the file is meant to be used by multiple APIs. Thanks for the explanation. We can keep this as it is.